state/protocol/params.go

```diff
@@ -23,6 +23,11 @@ type InstanceParams interface {
 	// No errors are expected during normal operation.
 	Root() (*flow.Header, error)
```

Comment:
would be great to change this to `FinalizedRoot` and update the documentation to explain how it is different from `SealedRoot`.